### PR TITLE
feature/CCS-775 UCA template values

### PR DESCRIPTION
**Pull request by saviogrossi**
- UCA.getUCAProps()
- UCA.parseValueFromProps()

### CODE REVIEW

File: __test__/UserCollectableAttributes.test.js [hunk 1]
```diff
@@ -401,4 +409,113 @@ describe('UCA Constructions tests', () => {
...
+    ];
+
+    const ucaValue = UCA.parseValueFromProps('cvc:Contact:email', propValues, '1');
+    expect(ucaValue.username).toEqual('savio');
+    expect(ucaValue.domain.name).toEqual('civic');
+    expect(ucaValue.domain.tld).toEqual('com');
+    function createUCA() {
+      return new UCA('cvc:Contact:email', ucaValue, '1');
+    }
+    expect(createUCA).not.toThrowError('name.first is not defined');
```
[the remark this answers is not in the document]
Comment (saviogrossi) [reply __test__/UserCollectableAttributes.test.js]: done, thks

File: README.md [hunk 2]
```diff
@@ -268,6 +268,79 @@ JSON String
 
 ```
 
+##### Getting UCA properties template and generation values
+
+```js
+const ucaTemplate = UCA.getUCAProps('cvc:Contact:email', '1');
+  //  ucaTemplate = {
```
[the remark this answers is not in the document]
Comment (saviogrossi) [reply README.md]: A key/value is better to serialize/deserialize for mobile Model Generation

File: src/UserCollectableAttribute.js [hunk 3]
```diff
@@ -165,6 +193,51 @@ class UserCollectableAttribute {
     return properties;
   }
 
+  static getUCAProps(identifier, version) {
+    const definition = version ? _.find(definitions, { identifier, version }) : _.find(definitions, { identifier });
+    if (!definition) {
+      return handleNotFoundDefinition(identifier, version);
```
[the remark this answers is not in the document]
Comment (saviogrossi) [reply src/UserCollectableAttribute.js]: Not sure. I just follow the getAllProperties() method in that case

File: src/UserCollectableAttribute.js [hunk 4]
```diff
@@ -165,6 +193,51 @@ class UserCollectableAttribute {
...
+    };
+
+    const typeDefinition = getTypeDefinition(definitions, identifier);
+
+    // If Object
+    if (typeDefinition && getTypeName(typeDefinition, definitions) === 'Object') {
+      const basePropertyName = getObjectBasePropName(definitions, typeDefinition);
+      ucaTemplate.basePropertyName = basePropertyName;
+    } else {
+      const identifierComponents = _.split(identifier, ':');
```
[the remark this answers is not in the document]
Comment (saviogrossi) [reply src/UserCollectableAttribute.js]: There are others, I will open a ticket for do this refactoring later.

File: src/UserCollectableAttribute.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (brunocarneiro) [reply src/UserCollectableAttribute.js]: If you look at the method declaration its first parameter is `definitions`.
`const handleNotFoundDefinition = (myDefinitions, identifier, version) => {...`

File: src/UserCollectableAttribute.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (brunocarneiro) [reply src/UserCollectableAttribute.js]: Alright

**Pull request merged**